### PR TITLE
Update javadocs

### PR DESCRIPTION
**Pull request by abhishekrb19**
#### Which Delta project/connector is this regarding?

- [ ] Spark
- [ ] Standalone
- [ ] Flink
- [x] Kernel
- [ ] Other (fill in here)

## Description
Changes:
- Fix a few Javadocs in the `Table` interface.
- Update `USER_GUIDE.md` to fix the optional vector usage.

## How was this patch tested?

N/A.

## Does this PR introduce _any_ user-facing changes?

Javadoc updates.

### CODE REVIEW

File: kernel/USER_GUIDE.md [hunk 1]
```diff
@@ -171,7 +171,7 @@ while(fileIter.hasNext()) {
           ColumnVector column0 = dataBatch.getColumnVector(0);
           for (int rowIndex = 0; rowIndex < column0.getSize(); rowIndex++) {
             // check if the row is selected or not
-            if (selectionVector.isEmpty() || selectionVector.get().getBoolean(rowIndex)) {
+            if (selectionVector.isPresent() && selectionVector.get().getBoolean(rowIndex)) {
```
Comment (vkorukanti) [inline kernel/USER_GUIDE.md:174]:
```suggestion
            if (!selectionVector.isPresent() ||
               (!selectionVector.get().isNullAt(rowId) && selectionVector.get().getBoolean(rowId)))  {
```

File: kernel/USER_GUIDE.md [hunk 2]
```diff
@@ -182,7 +182,7 @@ while(fileIter.hasNext()) {
 	  ColumnVector column1 = dataBatch.getColumnVector(1);
 	  for (int rowIndex = 0; rowIndex < column1.getSize(); rowIndex++) {
             // check if the row is selected or not
-            if (selectionVector.isEmpty() || selectionVector.get().getBoolean(rowIndex)) {
+            if (selectionVector.isPresent() && selectionVector.get().getBoolean(rowIndex)) {
```
Comment (vkorukanti) [inline kernel/USER_GUIDE.md:185]:
```suggestion
            if (!selectionVector.isPresent() ||
               (!selectionVector.get().isNullAt(rowId) && selectionVector.get().getBoolean(rowId)))  {
```

File: kernel/USER_GUIDE.md [hunk 3]
```diff
@@ -231,7 +231,7 @@ Optional<Predicate> remainingFilter = myFilteredScan.getRemainingFilter();
 
 The scan files returned by  `myFilteredScan.getScanFiles(myEngine)` will have rows representing files only of the required partition. Similarly, you can provide filters for non-partition columns, and if the data in the table is well clustered by those columns, then Delta Kernel will be able to skip files as much as possible.
 
-## Create to Delta table
+## Create Delta table
```
Comment (vkorukanti) [inline kernel/USER_GUIDE.md:234]:
```suggestion
## Create a Delta table
```

**Review by vkorukanti — COMMENTED** [review]
LGTM, could you please update the TOC? thank you!

**Review by vkorukanti — APPROVED** [review]

**Comment by abhishekrb19** [comment]
Thanks for the review, @vkorukanti!

**Pull request merged**